model/src/main/kotlin/VcsInfo.kt

```diff
@@ -78,11 +78,17 @@ data class VcsInfo(
         }
 
         var provider = this.provider
-        if (provider.equals(other.provider, true)) {
-            // Prefer the other provider only if its spelling matches the VCS class names.
-            if (other.provider.toLowerCase().capitalize() == other.provider) {
+        if (provider.isBlank()) {
+            if (other.provider.isNotBlank()) {
```

Comment:
Isn't this check superfluous? Setting the reference to another blank string is probably cheaper than checking if it is blank.

Reply:
Well, to be honest I only did split the `&&`into two nested `if`s in order to be able to use the `else` branch of the outer `if` 😆

Reply:
You're right, let's see...

Reply:
You're right that this check is not strictly needed. However, I didn't want to change this from the original code as part of this commit, as I would then also need to change similar occurrences below. Also, I guess reading memory two times is cheaper than reading and writing, so I'm actually not sure about it being cheaper performance-wise 😉